### PR TITLE
Add source_info to the tensors produced by video readers

### PR DESCRIPTION
**Pull request by JanuszL**
- adds missing source info to the outputs of the video reader and 
  experimental video reader

## Category:
**New feature** (*non-breaking change which adds functionality*)

## Description:
- adds missing source info to the outputs of the video reader and 
  experimental video reader
- related to https://github.com/NVIDIA/DALI/issues/4909

### Affected modules and functionalities:
- video reader
  - test_video.test_file_names_meta
  - test_video_pipeline.test_file_names_meta
- experimental video reader

### Tests:

- [ ] Existing tests apply
- [x] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

File: dali/test/python/test_video_pipeline.py [hunk 1]
```diff
@@ -198,6 +200,30 @@ def test_file_list_starts_ends_videopipeline():
...
 
+def test_file_names_meta():
+    files = []
+    for cont in video_types:
+        path = os.path.join(video_containers_data_root, cont)
+        files += [path + '/' + f for f in os.listdir(path)]
+    files = sorted(files)
+    list_file = tempfile.NamedTemporaryFile(mode="w", delete=False)
+    for f in files:
+        list_file.write("{} {} {} {}\n".format(os.path.join(VIDEO_DIRECTORY, f), 0, 0, 1))
```
Comment (stiepan) [inline dali/test/python/test_video_pipeline.py:211]: Aren't `files` already abolute paths to files inside subdir of the DALI_extra? In that case the `os.path.join(VIDEO_DIRECTORY, f)` would be just `f`?

**Review by stiepan — APPROVED** [review]

File: dali/test/python/test_video_pipeline.py (hunk 1 above)
Comment (JanuszL) [reply dali/test/python/test_video_pipeline.py]: True, fixed

File: dali/test/python/decoder/test_video.py [hunk 2]
```diff
@@ -185,3 +185,35 @@ def test_pipeline():
     for _ in range(iters):
         video_pipeline_0.run()
         video_pipeline_1.run()
+
+
+@params('cpu', 'gpu')
+def test_file_names_meta(device):
```
Comment (awolant) [inline dali/test/python/decoder/test_video.py:191]: Maybe the name of this test or some comment could mention `SourceInfo`? It took me a minute to figure out what is tested here?

**Review by awolant — APPROVED** [review]

File: dali/test/python/decoder/test_video.py (hunk 2 above)
Comment (JanuszL) [reply dali/test/python/decoder/test_video.py]: Done

**Comment by JanuszL** [comment]
!build

**Pull request merged**